homeassistant/components/husqvarna_automower/entity.py

```diff
@@ -0,0 +1,72 @@
+"""Platform for Husqvarna Automower basic entity."""
+
+import datetime
+import logging
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from . import AutomowerDataUpdateCoordinator
+from .const import DOMAIN, HUSQVARNA_URL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AutomowerEntity(CoordinatorEntity[AutomowerDataUpdateCoordinator]):
+    """Defining the Automower Basic Entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, coordinator, idx) -> None:
+        """Initialize AutomowerEntity."""
+        super().__init__(coordinator, context=idx)
+        self.idx = idx
+        self.mower = coordinator.session.data["data"][self.idx]
+        mower_attributes = self.get_mower_attributes()
+        self.mower_id = self.mower["id"]
+        self.mower_name = mower_attributes["system"]["name"]
+        self.model_name = mower_attributes["system"]["model"]
+
+        self._available = self.get_mower_attributes()["metadata"]["connected"]
+
+    def get_mower_attributes(self) -> dict:
+        """Get the mower attributes of the current mower."""
+        return self.coordinator.session.data["data"][self.idx]["attributes"]
+
+    def datetime_object(self, timestamp) -> datetime.datetime | None:
```

Comment:
I don't think this is a fitting place for this function. Would be better in a helpers.py or something along those lines. It could also be that there is a helper already in HA, but not that I know of.